src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -103,11 +107,13 @@ public static GroupNotification createNotification(Exercise exercise, User autho
             }
             case DUPLICATE_TEST_CASE -> {
                 title = DUPLICATE_TEST_CASE_TITLE;
-                text = "Exercise \"" + exercise.getTitle() + "\" has multiple test cases with the same name.";
+                text = notificationText;
```

Comment:
Is there a particular reason why only this notification gets text from the `notificationText`, but others have as hard coded string?

Reply:
Yes, there is one reason. I found out that there was a legacy final text for this notification but it has less information than the one which uses a (not null) notification text. This method (that generates a duplicate test case notification) is triggered only in one place and right before it always takes a different constant string and combines it with dynamic data, thus the notification text is always present. (I know that this breaks the convention a little bit. This is a bigger legacy issue, especially when it comes to EXERCISE_UPDATE notifications. This type should have multiple "subtypes" but this is/was realized via constant notification text strings. (e.g. I separated such a notification in [#4314])(https://github.com/ls1intum/Artemis/pull/4314)